dirwatcher.py

```diff
@@ -1,56 +1,92 @@
 __author__="Sasha Lukas + demo"
 
-import logging 
-import datetime
-import time
-import argparse
 import os
+import argparse
+import time
+import datetime
+import logging
+import signal
+
 logger = logging.getLogger(__file__)
+exit_flag = False
+files_found = []
+magic_word_position = {}
+
 
 def watch_directory(args):
-    watching_files = {}
-    logger.info('Watching directory: {}, File Ext: {}, Polling Interval: {}, Magic Text: {}'.format(
-    args.path, args.ext, args.interval, args.magic 
-    ))
-
-    # Keys are the actual filename and the values are where to begin searching.
-
-    # Look at directory and get a list of files from it.
-    # Add these to dictonary if not already present, and log it as a new file.
-    
-    # Look through your "watching_files" dictonary
-    # and compare that to a list of files that are in the dictonary.
-    
-    # If the file is not in your dictonary anymore you have
-    # to log the file and remove it from your dictonary
-
-    # Iterate through dictonary, open each file at the last line you read from, 
-    # Start reading from that point looking for any "magic" text
-    
-    # Update the last position that you read from in the dictionary  
-    
-    while True: 
-        try:
-            logger.info('Inside Watch Loop')
-            time.sleep(args.interval)
-        except KeyboardInterrupt:
-            break
+    """Watches given directory and reports when files matching the
+    given extension are added or removed.  Calls find_magic to search
+    present files for a given magic word"""
+    global files_found
+    global magic_word_position
+    logger.info('Watching directory: {}, File Ext: {}, Polling Interval: {}, '
```

Comment:
It's good to have this info message that shows the parameters of the program, but this message should be logged only once after the program has started running, but before it enters its polling loop.  Otherwise, a log file may fill up with stuff like this which is repetitive and gives no additional info
```
2020-04-20 21:30:45.949 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic Text: sasha
2020-04-20 21:30:47.951 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic Text: sasha
2020-04-20 21:30:49.952 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic Text: sasha
2020-04-20 21:30:51.954 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic Text: sasha
2020-04-20 21:30:53.958 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic Text: sasha
2020-04-20 21:30:55.959 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic Text: sasha
2020-04-20 21:30:57.961 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic Text: sasha
2020-04-20 21:30:59.963 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic Text: sasha
2020-04-20 21:31:01.964 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic Text: sasha
2020-04-20 21:31:03.966 dirwatcher.py INFO    [MainThread  ] Watching directory: watchme, File Ext: .log, Polling Interval: 2.0, Magic
```